### PR TITLE
Cleanup containers and images at the end of tests.

### PR DESCRIPTION
**Pull request by xingao267**

### CODE REVIEW

File: test/configs/autoconfig_test.sh [hunk 1]
```diff
@@ -30,9 +30,18 @@ WORKSPACE_ROOT=$(pwd)
...
 NAME=${TARGET##*/}
 DIR=${TARGET%${NAME}}
-
 autoconfig_script=${WORKSPACE_ROOT}/${DIR}${NAME}
 
+# Helper function for always delete the containers / temporary files on exit
+function cleanup_on_finish {
+  echo "=== Deleting images  ==="
+  docker images -a | grep "bazel/${DIR%/}" | awk '{print $3}' | xargs docker rmi -f
+  docker images -a | grep "test-" | awk '{print $3}' | xargs docker rmi -f
```
Comment (erain) [inline test/configs/autoconfig_test.sh]: For name-match in `docker ps` and deletion, we can do:
```
docker rm -f $(docker ps -f name=test- -q)
```

File: test/configs/debian8_clang_autoconfig_test.sh [hunk 2]
```diff
@@ -30,6 +30,15 @@ CONFIG_VERSION=0.3.0
 TEST_CONFIGS_DIR=${TEST_TMPDIR}/bazel-toolchains-${COMMIT}/configs/debian8_clang/${CONFIG_VERSION}/bazel_${BAZEL_VERSION}/
 AUTOCONFIG_SCRIPT=${WORKSPACE_ROOT}/rules/debian8-clang-${CONFIG_VERSION}-bazel_${BAZEL_VERSION}-autoconfig
 
+# Helper function for always delete the containers / temporary files on exit
+function cleanup_on_finish {
+  echo "=== Deleting images  ==="
+  docker images -a | grep "debian8-clang-${CONFIG_VERSION}-bazel_${BAZEL_VERSION}-autoconfig" | awk '{print $3}' | xargs docker rmi -f
```
Comment (erain) [inline test/configs/debian8_clang_autoconfig_test.sh]: Same as above, use `docker ps -f name=...` for name-match

File: test/configs/autoconfig_test.sh (hunk 1 above)
Comment (xingao267) [reply test/configs/autoconfig_test.sh]: The containers have already been deleted after the docker run command by using the --rm flag. I'm also sending a PR https://github.com/GoogleCloudPlatform/base-images-docker/pull/209 to delete container after install_pkgs rule. We are removing the images here.

Comment (erain) [reply test/configs/autoconfig_test.sh]: I see, it looks like `docker images -f` doesn't support `name` filter, so LGTM then.

File: test/configs/autoconfig_test.sh [hunk 3]
```diff
@@ -30,9 +30,24 @@ WORKSPACE_ROOT=$(pwd)
 TARGET=${TEST_BINARY%_test}
 NAME=${TARGET##*/}
 DIR=${TARGET%${NAME}}
-
 autoconfig_script=${WORKSPACE_ROOT}/${DIR}${NAME}
 
+# Helper function for always delete the containers / temporary files on exit
+function cleanup_on_finish {
+  echo "=== Deleting images  ==="
+  images=($(docker images -a | grep "bazel/${DIR%/}\|test-" | awk '{print $3}'))
```
[the remark this answers is not in the document]
Comment (xingao267) [reply test/configs/autoconfig_test.sh]: Using rbe-test now.

File: test/configs/debian8_clang_autoconfig_test.sh [hunk 4]
```diff
@@ -30,6 +30,22 @@ CONFIG_VERSION=0.3.0
 TEST_CONFIGS_DIR=${TEST_TMPDIR}/bazel-toolchains-${COMMIT}/configs/debian8_clang/${CONFIG_VERSION}/bazel_${BAZEL_VERSION}/
 AUTOCONFIG_SCRIPT=${WORKSPACE_ROOT}/rules/debian8-clang-${CONFIG_VERSION}-bazel_${BAZEL_VERSION}-autoconfig
 
+# Helper function for always delete the containers / temporary files on exit
+function cleanup_on_finish {
+  echo "=== Deleting images  ==="
+  images=($(docker images -a | grep "debian8-clang-${CONFIG_VERSION}-bazel_${BAZEL_VERSION}-autoconfig" | awk '{print $3}'))
```
[the remark this answers is not in the document]
Comment (xingao267) [reply test/configs/debian8_clang_autoconfig_test.sh]: I looked into the code a little more. And rules_docker hasn't exposed 'repository' to be overridden yet (https://github.com/bazelbuild/rules_docker/blob/master/container/image.bzl#L163).

I suggest we use the prefix now, and I've created a internal bug to modify our autoconfig rule, rules_docker, download_pkgs and install_pkgs to support getting a repository attribute. And the value should be passed down from the autoconfig rule.

[the remark this answers is not in the document]
Comment (nlopezgi) [reply test/configs/debian8_clang_autoconfig_test.sh]: can you change the "debian8-clang" for something else that can also be made generic for several targets (e.g., rbe-autoconfig-xxx)

[the remark this answers is not in the document]
Comment (nlopezgi) [reply test/configs/debian8_clang_autoconfig_test.sh]: spoke offline, this is only to delete the images created by the one target debian8-clang.... so looks fine

**Pull request merged**